src/Microsoft.Graph.Core/Exceptions/ErrorConstants.cs

```diff
@@ -20,11 +20,11 @@ internal static class Codes
 
             internal static string TooManyRedirects = "tooManyRedirects";
 
```

Comment:
Why have we removed this error?

Reply:
Since we are moving away from count-based retry handling to time-based, I saw it appropriate to have it replaced with:
https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/7b8c7644080acb4afb556875693421cbb0dcf782/src/Microsoft.Graph.Core/Exceptions/ErrorConstants.cs#L27

and

https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/7b8c7644080acb4afb556875693421cbb0dcf782/src/Microsoft.Graph.Core/Exceptions/ErrorConstants.cs#L66

Reply:
It wasn't my understanding that we were moving away from retry count based handling, we were simply adding another approach.  For operations that potentially take a varying amount of time to execute, time based retry may not be appropriate.

Reply:
@darrelmiller so this would ideally mean having an overloaded method for this: 
https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/2c23a8bdeeaa608e7e5e36494b443e5982fbef6b/src/Microsoft.Graph.Core/Requests/Middleware/RetryHandler.cs#L75-L114

1) for the **count-based** retry task, and
2) for the **retry-based** retry task

and check which option to use here:
https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/2c23a8bdeeaa608e7e5e36494b443e5982fbef6b/src/Microsoft.Graph.Core/Requests/Middleware/RetryHandler.cs#L61-L64

Ideas are welcomed @peombwa

Reply:
That would create a lot of redundancy.  I would try and add a condition to the while loop.  The max retry condition should stay and we should add another condition that effectively says "if someone has specified a retriesTimeLimit and we won't exceed the time limit with with this retry then let's do it".

Reply:
@darrelmiller in essence you are saying that the count-based retry condition should be run side-by-side with the time-based retry condition? Won't one cancel out the other in certain scenarios before success? e.g. 3 retries of 1 sec. where a user specified a retriesTimeLimit of 10 sec. would probably receive a failure before reaching the 10 sec. time limit? or 3 retries of 10 sec. where a user specified a retriesTimeLimit of 5 sec. would also expect a failure before counting down the full 3 retries?

Reply:
Which ever limit is hit first should be respected. So, with your example, and assuming a max retry count of 3, the retry count limit would be hit before the max cumulative time limit.